mirascope/chat/utils.py

```diff
@@ -51,27 +54,41 @@ def convert_function_to_openai_tool(fn: Callable) -> Type[OpenAITool]:
         doc += "\n\n" + docstring.long_description
 
     field_definitions = {}
+    hints = get_type_hints(fn)
     for i, parameter in enumerate(signature(fn).parameters.values()):
+        if parameter.name == "self" or parameter.name == "cls":
+            continue
         if parameter.annotation == Parameter.empty:
             raise ValueError("All parameters must have a type annotation.")
 
-        docstring_param = docstring.params[i]
-        if docstring_param.arg_name != parameter.name:
-            raise ValueError(
-                f"Function parameter name {parameter.name} does not match docstring "
-                f"parameter name {docstring_param.arg_name}. Make sure that the "
-                "parameter names match exactly."
-            )
-        if not docstring_param.description:
-            raise ValueError("All parameters must have a description.")
-
-        field_definitions[parameter.name] = (
-            parameter.annotation,
-            FieldInfo(
-                description=docstring.params[i].description, default=parameter.default
-            )
-            if parameter.default != Parameter.empty
-            else FieldInfo(description=docstring.params[i].description),
+        param_description = None
+        if i < len(docstring.params):
+            docstring_param = docstring.params[i]
+            if docstring_param.arg_name != parameter.name:
+                raise ValueError(
+                    f"Function parameter name {parameter.name} does not match docstring "
+                    f"parameter name {docstring_param.arg_name}. Make sure that the "
+                    "parameter names match exactly."
+                )
+            if not docstring_param.description:
+                raise ValueError("All parameters must have a description.")
+            param_description = docstring_param.description
+
+        field_info_kwargs = {"annotation": hints[parameter.name]}
+        if parameter.default != Parameter.empty:
```

Comment:
Add some comments, why does default need to check empty but description can check falsy

Reply:
Done.